jetstream_pt/quantize.py

```diff
@@ -95,3 +97,63 @@ def load_q_weight_helper(w_q, scale, zp=None, block_size=-1):
       zp = (zp * scale).transpose(1, 0).squeeze(-1).to(torch.bfloat16)
     scale = scale.transpose(1, 0).squeeze(-1).to(torch.bfloat16)
   return w_q, scale, zp
+
+
+def blockwise_jax_kernel(inputs, weight, weight_scaler, zero_point):
+  """Blockwise Matmul kernel impl in JAX using einsum"""
+  weight = weight.astype(jnp.int8)
+  block_size = weight.shape[1]
+  inputs_shape = inputs.shape
+  inputs_new_shape = inputs_shape[:-1] + (
+      inputs_shape[-1] // block_size,
+      block_size,
+  )
+  inputs = inputs.reshape(inputs_new_shape)
+  out = jnp.einsum("scz,bdsc->bdsz", weight, inputs)
+  out = jnp.einsum("bdsz,sz->bdz", out, weight_scaler)
+  if zero_point is not None:
+    zp_out = jnp.einsum("bdsc,sz->bdz", inputs, zero_point)
+    out = out - zp_out
+  return out
+
+
+def blockwise_jax_kernel_dot_general(inputs, weight, weight_scaler, zero_point):
```

Comment:
No, we don't need to call jax for it. Thanks for the heads up. Since I'm moving the existing kernel implementation to this new file, I will switch to torch in the following PR.